src/log4net/Appender/AdoNetAppender.cs

```diff
@@ -553,8 +553,17 @@ protected virtual void SendBuffer(IDbTransaction dbTran, LoggingEvent[] events)
 					{
 						dbCmd.Transaction = dbTran;
 					}
-					// prepare the command, which is significantly faster
-					dbCmd.Prepare();
+
+					try
+					{
+						// prepare the command, which is significantly faster
+						dbCmd.Prepare();
+					}
+					catch (Exception)
+					{
+						// Ignore exception
```

Comment:
whilst I'd like to track down the _why_ of this instead of skipping over it, I think that may take some time -- I don't even have pgsql installed on any of my machines. As far as I understand, the only side-effect here is that the command might run a little slower, so I'd say, let's put in this workaround, but please put in a comment explaining why, eg `// ignore prepare exceptions as they can happen without affecting actual logging, eg on pgsql`. Thoughts?

Reply:
I've investigated it. The simple _why_ is to prepare a valid command instead of invalid one without parameters and values.
Also tracked down to inner command cache, so it is not necessary to prepare command.